modules/core/src/smithy4s/http/internals/package.scala

```diff
@@ -52,4 +52,23 @@ package object internals {
       }
   }
 
+  private[smithy4s] def pathSegments(
+      str: String
+  ): Option[Vector[PathSegment]] = {
+    str
+      .split('/')
+      .toVector
+      .filterNot(_.isEmpty())
+      .traverse(fromToString(_))
```

Comment:
...we have traverse? cool

Reply:
poor man's traverse : I've implemented it for `vector + option` in the `http.internals` package

Reply:
we have traverse at home!